### PR TITLE
:sparkles: Source GitHub: added Issue timeline events stream

### LINKED ISSUE #12316: Source Github: add `timeline` stream
## Tell us about the problem you're trying to solve
The Github connector has support for the [Github Issue Events API](https://docs.github.com/en/developers/webhooks-and-events/events/issue-event-types) but it doesn't seem to support data ingested from the [Github Issue Timeline Events API](https://docs.github.com/en/rest/issues/timeline). This means, for example, that `reviewed` events are missing on pull requests.

## Describe the solution you’d like
Either a new stream, or an addition to the existing Issue Events stream to support Timeline events.

## Describe the alternative you’ve considered or used
The alternative is to not have the data, which doesn't seem ideal.

## Additional context
As an example: [reviewed event](https://docs.github.com/en/developers/webhooks-and-events/events/issue-event-types#reviewed) isn't available on the events API, just the timeline events API.

## Are you willing to submit a PR?
I'm not sure I have the time to do so at the moment, as I haven't looked at creating a connector before. Perhaps I can look at it if no-one picks it up.

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What
resolved: https://github.com/airbytehq/airbyte/issues/12316
API doc: https://docs.github.com/en/rest/issues/timeline?apiVersion=2022-11-28

## How
Added stream Issue Timeline Events

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`db246b63d4`](https://github.com/airbytehq/airbyte/pull/30823/commits/db246b63d4b2c42a84bcec4188b4875c4fde9b7d)) - ❌

⏲️  Total pipeline duration: 06mn04s 

| Step                                                       | Result   |
|:-----------------------------------------------------------|:---------|
| Connector package install                                  | ✅        |
| Build source-github docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                 | ✅        |
| Acceptance tests                                           | ✅        |
| Code format checks                                         | ✅        |
| Validate metadata for source-github                        | ✅        |
| Connector version semver check                             | ✅        |
| Connector version increment check                          | ❌        |
| QA checks                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-github_timeline-events-stream/1695899963/db246b63d4b2c42a84bcec4188b4875c4fde9b7d/source-github/1.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:db246b63d4b2c42a84bcec4188b4875c4fde9b7d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`b5b4cb0333`](https://github.com/airbytehq/airbyte/pull/30823/commits/b5b4cb033325b033d32f2dc8cdaebfbcc9bbab01)) - ❌

⏲️  Total pipeline duration: 04mn56s 

| Step                                                       | Result   |
|:-----------------------------------------------------------|:---------|
| Connector package install                                  | ✅        |
| Build source-github docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                 | ✅        |
| Acceptance tests                                           | ❌        |
| Code format checks                                         | ✅        |
| Validate metadata for source-github                        | ✅        |
| Connector version semver check                             | ✅        |
| Connector version increment check                          | ✅        |
| QA checks                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-github_timeline-events-stream/1695902251/b5b4cb033325b033d32f2dc8cdaebfbcc9bbab01/source-github/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:b5b4cb033325b033d32f2dc8cdaebfbcc9bbab01)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`712c0e3487`](https://github.com/airbytehq/airbyte/pull/30823/commits/712c0e34870cc20a66368230ccdb0da945072eaa)) - ✅

⏲️  Total pipeline duration: 06mn23s 

| Step                                                       | Result   |
|:-----------------------------------------------------------|:---------|
| Connector package install                                  | ✅        |
| Build source-github docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                 | ✅        |
| Acceptance tests                                           | ✅        |
| Code format checks                                         | ✅        |
| Validate metadata for source-github                        | ✅        |
| Connector version semver check                             | ✅        |
| Connector version increment check                          | ✅        |
| QA checks                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-github_timeline-events-stream/1695906599/712c0e34870cc20a66368230ccdb0da945072eaa/source-github/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:712c0e34870cc20a66368230ccdb0da945072eaa)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

File: docs/integrations/sources/github.md [hunk 1]
```diff
@@ -165,9 +166,10 @@ The GitHub connector should not run into GitHub API limitations under normal usa
...
-| 1.3.0   | 2023-08-25 | [30731](https://github.com/airbytehq/airbyte/pull/30731)                                                          | Add new stream `ProjectsV2`                                                                                                                                         |
-| 1.2.1   | 2023-08-22 | [30693](https://github.com/airbytehq/airbyte/pull/30693)                                                          | Handle 404 error in `TeamMemberShips`                                                                                                                               |
-| 1.2.0   | 2023-08-22 | [30647](https://github.com/airbytehq/airbyte/pull/30647)                                                          | Add support for self-hosted GitHub instances                                                                                                                        |
+| 1.4.0   | 2023-09-28 | [30823](https://github.com/airbytehq/airbyte/pull/30823)              | Add new stream `issue Timeline Events`                                                                                                                              |
```
Comment (davydov-d) [inline docs/integrations/sources/github.md:169]: please format the table

File: airbyte-integrations/connectors/source-github/integration_tests/configured_catalog.json [hunk 2]
```diff
@@ -420,6 +420,16 @@
       },
       "sync_mode": "full_refresh",
       "destination_sync_mode": "overwrite"
+    },
+    {
+      "stream": {
+        "name": "issue_timeline_events",
+        "json_schema": {},
+        "supported_sync_modes": ["full_refresh"],
+        "source_defined_primary_key": []
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-github/integration_tests/configured_catalog.json:429]: please add the primary key here

**Review by davydov-d — APPROVED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`5dae51d781`](https://github.com/airbytehq/airbyte/pull/30823/commits/5dae51d78116e5de791a9c186901eaf6f854c1aa)) - ✅

⏲️  Total pipeline duration: 03mn27s 

| Step                                                       | Result   |
|:-----------------------------------------------------------|:---------|
| Connector package install                                  | ✅        |
| Build source-github docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                 | ✅        |
| Acceptance tests                                           | ✅        |
| Code format checks                                         | ✅        |
| Validate metadata for source-github                        | ✅        |
| Connector version semver check                             | ✅        |
| Connector version increment check                          | ✅        |
| QA checks                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-github_timeline-events-stream/1695910056/5dae51d78116e5de791a9c186901eaf6f854c1aa/source-github/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:5dae51d78116e5de791a9c186901eaf6f854c1aa)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`918f854dff`](https://github.com/airbytehq/airbyte/pull/30823/commits/918f854dff9e8c3da56dfbb12e9737ae000b2389)) - ✅

⏲️  Total pipeline duration: 06mn18s 

| Step                                                       | Result   |
|:-----------------------------------------------------------|:---------|
| Connector package install                                  | ✅        |
| Build source-github docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                 | ✅        |
| Acceptance tests                                           | ✅        |
| Code format checks                                         | ✅        |
| Validate metadata for source-github                        | ✅        |
| Connector version semver check                             | ✅        |
| Connector version increment check                          | ✅        |
| QA checks                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-github_timeline-events-stream/1695910906/918f854dff9e8c3da56dfbb12e9737ae000b2389/source-github/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:918f854dff9e8c3da56dfbb12e9737ae000b2389)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`911697bac3`](https://github.com/airbytehq/airbyte/pull/30823/commits/911697bac349351295369328bda7a7afa6d8c50d)) - ✅

⏲️  Total pipeline duration: 03mn32s 

| Step                                                       | Result   |
|:-----------------------------------------------------------|:---------|
| Connector package install                                  | ✅        |
| Build source-github docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                 | ✅        |
| Acceptance tests                                           | ✅        |
| Code format checks                                         | ✅        |
| Validate metadata for source-github                        | ✅        |
| Connector version semver check                             | ✅        |
| Connector version increment check                          | ✅        |
| QA checks                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-github_timeline-events-stream/1695912656/911697bac349351295369328bda7a7afa6d8c50d/source-github/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:911697bac349351295369328bda7a7afa6d8c50d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`7106c8376f`](https://github.com/airbytehq/airbyte/pull/30823/commits/7106c8376f3cc313ec4b7e9916541d633b810126)) - ✅

⏲️  Total pipeline duration: 06mn05s 

| Step                                                       | Result   |
|:-----------------------------------------------------------|:---------|
| Connector package install                                  | ✅        |
| Build source-github docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                 | ✅        |
| Acceptance tests                                           | ✅        |
| Code format checks                                         | ✅        |
| Validate metadata for source-github                        | ✅        |
| Connector version semver check                             | ✅        |
| Connector version increment check                          | ✅        |
| QA checks                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-github_timeline-events-stream/1695980537/7106c8376f3cc313ec4b7e9916541d633b810126/source-github/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:7106c8376f3cc313ec4b7e9916541d633b810126)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

**Pull request merged**